src/main.rs

```diff
@@ -96,35 +106,48 @@ fn highlight(q: Json<Query>) -> JsonValue {
             // name. This is done due to some syntaxes matching an "extension"
             // that is actually a whole file name (e.g. "Dockerfile" or "CMakeLists.txt")
             // see https://github.com/trishume/syntect/pull/170
-            match syntax_set.find_syntax_by_extension(file_name) {
-                Some(v) => v,
-                None => 
-                    // Now try to find the syntax by the actual file extension.
-                    match syntax_set.find_syntax_by_extension(extension) {
-                        Some(v) => v,
-                        None =>
-                            // Fall back: Determine syntax definition by first line.
-                            match syntax_set.find_syntax_by_first_line(&q.code) {
-                                Some(v) => v,
-                                None => {
-                                    is_plaintext = true;
-
-                                    // Render plain text, so the user gets the same HTML
-                                    // output structure.
-                                    syntax_set.find_syntax_plain_text()
-                                }
-                        },
-                    }
-            }
+            syntax_set
+                .find_syntax_by_extension(file_name)
+                .or_else(|| syntax_set.find_syntax_by_extension(extension))
+                .or_else(|| syntax_set.find_syntax_by_first_line(&q.code))
```

Comment:
Hmmm, this `or_else` logic _sounds_ smart at first glance but I'm actually not so sure about it. I think this could well make it harder in practice to figure out why a language is being highlighted a certain way in the case of the wrong language being picked up.

Let's try it and see, but maybe something to revisit later.

Reply:
Fair enough. Definitely not attached to it